src/naive_bayes.cpp

```diff
@@ -0,0 +1,18 @@
+// -*- mode: C++; c-indent-level: 4; c-basic-offset: 4; indent-tabs-mode: nil; -*-
+
+#include <RcppMLPACK.h>				// MLPACK, Rcpp and RcppArmadillo
+
+#include <mlpack/methods/naive_bayes/naive_bayes_classifier.hpp> 	// particular algorithm used here
+
+// [[Rcpp::export]]
+Rcpp::List naive_bayes(const arma::mat& train, const arma::mat& test, const arma::Row<size_t>& labels, const int& classes) {
+    
+    arma::Row<size_t> results;
+
+    // Initialize with the default arguments.
+    mlpack::naive_bayes::NaiveBayesClassifier<> nbc(train, labels, classes);
+    
+    nbc.Classify(test, results);
+
+    return Rcpp::List::create(Rcpp::Named("results") = results);
```

Comment:
Ok, I played with this some more.  We were not thinking straight.

`arma::row` and `arma::col` are aliases for the default of `double`.  But `size_t` is `unsigned long`.  So this maybe aliased to `arma:ivec` or something else to give us integers.

You will have to cast this in any event because R only has `integer` and `numeric` (aka double).